### PR TITLE
Add verify task to deployment scripts

### PR DESCRIPTION
**Pull request by thongtrungtran**
- upgrade hardhat and hardhat-etherscan
- upgrade nodejs for travis
- set initialBaseFeePerGas=0 to hardhat config

### CODE REVIEW

**Review by Zohar97 — CHANGES_REQUESTED** [review]
- add export contract address to file after deploy
- add more network on config hardhat
- update duplicate await sleep(sleepTime) in helper + move sleepTime to env
- add env.example

**Review by Zohar97 — APPROVED** [review]

File: deployment/deployZapIn.js [hunk 1]
```diff
@@ -1,7 +1,9 @@
+const {runVerifyAndExport} = require('./helper');
+
 const ZapIn = artifacts.require('ZapIn');
 
-async function main () {
-  // ropsten 
+async function main() {
+  // ropsten
   const wethAddress = '0xc778417e063141139fce010982780140aa0cd5ab';
```
Comment (manhlx3006) [inline deployment/deployZapIn.js]: should get the address from some configs or files, sometimes I forget to update the weth or factory address and deploy for other networks

Comment (thongtrungtran) [reply deployment/deployZapIn.js]: upgrade to use **hardhat-deploy** for better development

**Comment by Zohar97** [comment]
- should move constant address or config to new folder which named deployConfig, then we can read address from there instead of read directly from hardhat.config file
- add deploy script example to readme

**Review by Zohar97 — APPROVED** [review]

File: deployConfigs/dmmFactory.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  "bsc": "0x878dFE971d44e9122048308301F540910Bbd934c",
+  "avax_testnet": "0x878dFE971d44e9122048308301F540910Bbd934c",
+  "avax": "0x10908C875D865C66f271F5d3949848971c9595C9",
+  "ftm_testnet": "0x10908C875D865C66f271F5d3949848971c9595C9",
+  "ftm": "0x78df70615ffc8066cc0887917f2Cd72092C86409",
+  "cronos": "0xD9bfE9979e9CA4b2fe84bA5d4Cf963bBcB376974",
+  "aurora": "0xD9bfE9979e9CA4b2fe84bA5d4Cf963bBcB376974",
+  "arbitrum": "0xD9bfE9979e9CA4b2fe84bA5d4Cf963bBcB376974",
+  "bttc": "0xD9bfE9979e9CA4b2fe84bA5d4Cf963bBcB376974",
+  "oasis": "0xD9bfE9979e9CA4b2fe84bA5d4Cf963bBcB376974"
```
Comment (manhlx3006) [inline deployConfigs/dmmFactory.json]: I think we also have Velas

Comment (manhlx3006) [reply deployConfigs/dmmFactory.json]: The factory could be different for different versions, like new Classic protocol should have the same Factory contract across all networks

Comment (thongtrungtran) [reply deployConfigs/dmmFactory.json]: Will add different address for different versions according to engineering wiki docs

Comment (thongtrungtran) [reply deployConfigs/dmmFactory.json:16]: btw, the address that is used across all networks(or almost networks) is defined by field "default" in the config files

**Review by Zohar97 — APPROVED** [review]

**Review by manhlx3006 — APPROVED** [review]

**Pull request merged**